src/detect-bsize.c

```diff
@@ -266,6 +268,43 @@ static DetectBsizeData *DetectBsizeParse (const char *str)
     return bsz;
 }
 
+static bool DetectBsizeCheckContent(Signature *s, int list, DetectBsizeData *bsz)
+{
+    SigMatch *sm = s->init_data->smlists[list];
+    for (; sm != NULL; sm = sm->next) {
+        if (sm->type != DETECT_CONTENT)
+            continue;
+
+        DetectContentData *cd = (DetectContentData *) sm->ctx;
+        SCLogNotice("Content %.*s, content length %"PRIu16, cd->content_len, cd->content, cd->content_len);
+        bool possible;
+
+        switch (bsz->mode) {
+            case DETECT_BSIZE_EQ:
+                possible = bsz->lo == cd->content_len;
```

Comment:
just checking for content_len is too limiting. What happens if we have:
`http.uri; content:"abc"; offset:3; depth:3; bsize:3;`? The sig can't match because we need 6 bytes of data.